New Serverless Pattern - Receive Email using Amazon SES (#2106)
aws-samples/serverless-patterns: ses-lambda-sam/README.md

   - Leave rest of the options as default.

5. Note the outputs from the SAM deployment process. This contain the resource names and/or Ids which are used for next step as well as for testing.

6. From the command line, please execute the below command. Please replace `EmailReceiptRuleSet` from the `sam deploy` output and also your region. This will activate the Amazon SES ruleset to invoke the AWS Lambda function once it receives the email.
   ```bash
   aws ses set-active-receipt-rule-set  --rule-set-name {EmailReceiptRuleSet} --region {your-region}
   ```

7. By now, you should receive an email confirmation with subject line `DKIM setup SUCCESS for {DomainName} in {your-region}`. 

## How it works

Please refer to the architecture diagram below:

![End to End Architecture](images/architecture.png)

* User sends email to the configured domain in Amazon SES.
* Amazon SES receives the email and then applies the rule set to trigger an AWS Lambda function. 

> * The AWS Lambda functoin processes information from the SES event.

Reviewer: ```suggestion
* The AWS Lambda function processes the information from the SES event.
```

Author: Fixed.

## Testing

1. From the command line, please execute the below command to tail logs of the AWS Lambda function. Please replace `FunctionName` from the `sam deploy` output and also your region. 
   ```bash
   aws logs tail --follow /aws/lambda/{FunctionName} --region {your-region}
   ```

> 2. Make a note of `ReceivingEmailAddress` from the `sam deploy` output and send email to that email id. Send an email to this Id.

Reviewer: ```suggestion
2. Make a note of `ReceivingEmailAddress` from the `sam deploy` output and send email to that address.
```

> 3. You should see logline similar to following in the Lambda function log: 

Reviewer: Is this in near real-time or should the user expect some delay?

Author: It is event-driven. For me it took <5 sec. So I have added line to expect a few seconds of delay.
